compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FileClassLowering.kt

```diff
@@ -79,27 +81,36 @@ private class FileClassLowering(val context: JvmBackendContext) : FileLoweringPa
 
     private fun createFileClass(irFile: IrFile, fileClassMembers: List<IrDeclaration>): IrClass {
         val fileEntry = irFile.fileEntry
-        val fileClassInfo: JvmFileClassInfo?
-        when (fileEntry) {
+        val fileClassInfo = when (fileEntry) {
             is PsiSourceManager.PsiFileEntry -> {
                 val ktFile = context.psiSourceManager.getKtFile(fileEntry)
                     ?: throw AssertionError("Unexpected file entry: $fileEntry")
-                fileClassInfo = JvmFileClassUtil.getFileClassInfoNoResolve(ktFile)
+                JvmFileClassUtil.getFileClassInfoNoResolve(ktFile)
             }
             is NaiveSourceBasedFileEntryImpl -> {
-                fileClassInfo = JvmSimpleFileClassInfo(PackagePartClassUtils.getPackagePartFqName(irFile.fqName, fileEntry.name), false)
+                JvmSimpleFileClassInfo(PackagePartClassUtils.getPackagePartFqName(irFile.fqName, fileEntry.name), false)
             }
             else -> error("unknown kind of file entry: $fileEntry")
         }
         val isMultifilePart = fileClassInfo.withJvmMultifileClass
+
+        val onlyPrivateDeclarationsAndFeatureIsEnabled =
+            context.state.languageVersionSettings.supportsFeature(LanguageFeature.PackagePrivateFileClassesWithAllPrivateMembers) && fileClassMembers
+                .filterIsInstance<IrDeclarationWithVisibility>()
+                .map(IrDeclarationWithVisibility::visibility)
+                .all(DescriptorVisibilities::isPrivate)
```

Comment:
It would be great to also rewrite this to that it wouldn't create extra collections

Reply:
Can't I just add asSequence call?